### PR TITLE
Publish a build scan from Jenkins build

### PR DESCRIPTION
**Pull request by swegner**
Publish a Gradle [Build Scan](https://scans.gradle.com/get-started) from Jenkins builds. This is useful for debugging failures.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by swegner** [comment]
R: @lukecwik 

See how scan gets integrated into Jenkins page here: https://builds.apache.org/job/beam_PreCommit_Go_GradleBuild/1912/

**Comment by swegner** [comment]
Ping: @lukecwik please take a look

**Comment by swegner** [comment]
R: @aaltay

**Comment by swegner** [comment]
FYI, build scans are also enabled in the `apache/groovy` build: [`build.gradle`](https://github.com/apache/groovy/blob/c1bf7b56a9b4c9ccfd86a49e8546eef969bcbb9e/build.gradle#L57)

This is enabled only on Jenkins machines, so I see no issue with accepting TOS on developers' behalf. In the same way that there's no explicit opt-in for Jenkins or Coveralls.

File: build.gradle [hunk 1]
```diff
@@ -65,6 +66,19 @@ buildscript {
 
 apply plugin: "base"
 
+apply plugin: "com.gradle.build-scan"
+// JENKINS_HOME and BUILD_ID set automatically during Jenkins execution
+def isCIBuild = ['JENKINS_HOME', 'BUILD_ID'].every System.&getenv
+if (isCIBuild) {
+  buildScan {
+    // Build Scan enabled and TOS accepted for Jenkins lab build. This does not apply to builds on
```
Comment (aaltay) [inline build.gradle]: Please make sure to bring this to the attention of dev@.

**Pull request merged**